### PR TITLE
Migrations for new Guid property on Project, McPkg and CommPkg

### PR DESCRIPTION
**Pull request by terjewoldseth**
Updatet constructor and test for Project

### CODE REVIEW

**Review by arildbjerkenes — APPROVED** [review]

File: src/Equinor.ProCoSys.IPO.Command/CommPkgCommands/FillPcsGuids/FillCommPkgPcsGuidsCommand.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
+﻿using MediatR;
+using ServiceResult;
+
+namespace Equinor.ProCoSys.IPO.Command.CommPkgCommands.FillCommPkgPcsGuids
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/CommPkgCommands/FillPcsGuids/FillCommPkgPcsGuidsCommand.cs:4]: ns mismatch folder

File: src/Equinor.ProCoSys.IPO.Command/CommPkgCommands/FillPcsGuids/FillCommPkgPcsGuidsCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+using Equinor.ProCoSys.IPO.Domain;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.ProjectAggregate;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using ServiceResult;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.InvitationAggregate;
+using Equinor.ProCoSys.IPO.ForeignApi.MainApi.CommPkg;
+using System.Collections.Generic;
+
+namespace Equinor.ProCoSys.IPO.Command.CommPkgCommands.FillCommPkgPcsGuids
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/CommPkgCommands/FillPcsGuids/FillCommPkgPcsGuidsCommandHandler.cs:15]: ns mismatch folder

File: src/Equinor.ProCoSys.IPO.Command/McPkgCommands/FillPcsGuids/FillMcPkgPcsGuidsCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+using MediatR;
+using Microsoft.Extensions.Logging;
+using ServiceResult;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.InvitationAggregate;
+using Equinor.ProCoSys.IPO.ForeignApi.MainApi.CommPkg;
+using System.Collections;
+using System.Collections.Generic;
+using Equinor.ProCoSys.IPO.ForeignApi.MainApi.McPkg;
+
+namespace Equinor.ProCoSys.IPO.Command.McPkgCommands.FillMcPkgPcsGuids
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/McPkgCommands/FillPcsGuids/FillMcPkgPcsGuidsCommandHandler.cs:18]: ns mismatch folder

File: src/Equinor.ProCoSys.IPO.Command/McPkgCommands/FillPcsGuids/FillMcPkgPcsGuidsCommand.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using MediatR;
+using ServiceResult;
+
+namespace Equinor.ProCoSys.IPO.Command.McPkgCommands.FillMcPkgPcsGuids
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/McPkgCommands/FillPcsGuids/FillMcPkgPcsGuidsCommand.cs:4]: ns mismatch folder

File: src/Equinor.ProCoSys.IPO.Command/McPkgCommands/FillPcsGuids/FillMcPkgPcsGuidsCommandHandler.cs [hunk 5]
```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Equinor.ProCoSys.Common.Misc;
+using Equinor.ProCoSys.IPO.Domain;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.ProjectAggregate;
+using global::Equinor.ProCoSys.IPO.ForeignApi.MainApi.Project;
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/McPkgCommands/FillPcsGuids/FillMcPkgPcsGuidsCommandHandler.cs:8]: remove unused usings

File: src/Equinor.ProCoSys.IPO.Command/McPkgCommands/FillPcsGuids/FillMcPkgPcsGuidsCommandHandler.cs [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+using System.Collections.Generic;
+using Equinor.ProCoSys.IPO.ForeignApi.MainApi.McPkg;
+
+namespace Equinor.ProCoSys.IPO.Command.McPkgCommands.FillMcPkgPcsGuids
+{
+    public class FillPCSGuidsCommandHandler : IRequestHandler<FillMcPkgPCSGuidsCommand, Result<Unit>>
+    {
+        private readonly ILogger<FillMcPkgPCSGuidsCommand> _logger;
+        private readonly IInvitationRepository _invitationRepository;
+        private readonly ICommPkgApiService _commPkgApiService;
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/McPkgCommands/FillPcsGuids/FillMcPkgPcsGuidsCommandHandler.cs:24]: remove, not used

File: src/Equinor.ProCoSys.IPO.Command/McPkgCommands/FillPcsGuids/FillMcPkgPcsGuidsCommandHandler.cs [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+                    {
+                        var mcPkgDetails = await _mcPkgApiService.GetMcPkgByIdAsync(_plantProvider.Plant, mcPkgWithId.First().Id);
+                        mcPkg.Guid = mcPkgDetails.ProCoSysGuid;
+                       _logger.LogInformation($"FillMcPkgPCSGuids: McPkg updated: {mcPkg.McPkgNo}");
+                       count++;
+                    }
+                }
+            }
+
+          if (request.SaveChanges && count > 0)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/McPkgCommands/FillPcsGuids/FillMcPkgPcsGuidsCommandHandler.cs:72]: indent

File: src/Equinor.ProCoSys.IPO.Command/ProjectCommands/FillPcsGuids/FillProjectPcsGuidsCommandHandler.cs [hunk 8]
```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Linq;
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/ProjectCommands/FillPcsGuids/FillProjectPcsGuidsCommandHandler.cs:2]: remove unused usings

File: src/Equinor.ProCoSys.IPO.Command/ProjectCommands/FillPcsGuids/FillProjectPcsGuidsCommand.cs [hunk 9]
```diff
@@ -0,0 +1,12 @@
+﻿using MediatR;
+using ServiceResult;
+
+namespace Equinor.ProCoSys.IPO.Command.ProjectCommands.FillProjectPcsGuids
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/ProjectCommands/FillPcsGuids/FillProjectPcsGuidsCommand.cs:4]: ns mismatch folder

File: src/Equinor.ProCoSys.IPO.Command/ProjectCommands/FillPcsGuids/FillProjectPcsGuidsCommandHandler.cs [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.ProjectAggregate;
+using MediatR;
+using Microsoft.Extensions.Logging;
+using ServiceResult;
+using Equinor.ProCoSys.IPO.Domain.AggregateModels.InvitationAggregate;
+using Equinor.ProCoSys.IPO.ForeignApi.MainApi.CommPkg;
+using System.Collections.Generic;
+using Equinor.ProCoSys.IPO.ForeignApi.MainApi.Project;
+
+namespace Equinor.ProCoSys.IPO.Command.ProjectCommands.FillProjectPcsGuids
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/ProjectCommands/FillPcsGuids/FillProjectPcsGuidsCommandHandler.cs:16]: ns mismatch folder

File: src/Equinor.ProCoSys.IPO.Command/ProjectCommands/FillPcsGuids/FillProjectPcsGuidsCommandHandler.cs [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+                    if (projectDetails != null)
+                    {
+                        project.Guid = projectDetails.ProCoSysGuid;
+                       _logger.LogInformation($"FillProjectPCSGuids: Project updated: {project.Name}");
+                       count++;
+                    }
+                }
+            }
+
+          if (request.SaveChanges && count > 0)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Command/ProjectCommands/FillPcsGuids/FillProjectPcsGuidsCommandHandler.cs:62]: indent

File: src/Equinor.ProCoSys.IPO.Domain/AggregateModels/InvitationAggregate/CommPkg.cs [hunk 12]
```diff
@@ -63,6 +65,7 @@ protected CommPkg()
         public int ProjectId { get; private set; }
         public bool RfocAccepted { get; set; }
         public ICollection<Certificate> CertificateScopes => _certificateScope;
+        public Guid Guid { get; set; } = Guid.Empty;
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Domain/AggregateModels/InvitationAggregate/CommPkg.cs:68]: remove init  Guid.Empty;
Add a Todo // make this private after FillGuids is completed

File: src/Equinor.ProCoSys.IPO.Domain/AggregateModels/ProjectAggregate/Project.cs [hunk 13]
```diff
@@ -31,6 +32,7 @@ public Project(string plant, string name, string description)
         public int CreatedById { get; private set; }
         public DateTime? ModifiedAtUtc { get; private set; }
         public int? ModifiedById { get; private set; }
+        public Guid Guid { get; set; } = Guid.Empty;
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Domain/AggregateModels/ProjectAggregate/Project.cs:35]: remove init Guid.Empty;
Add a Todo // make Guid private after FillGuids is completed

File: src/Equinor.ProCoSys.IPO.Domain/AggregateModels/InvitationAggregate/CommPkg.cs
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Domain/AggregateModels/InvitationAggregate/CommPkg.cs:1]: Remove property SystemSubString. Not in use

File: src/Equinor.ProCoSys.IPO.Domain/AggregateModels/InvitationAggregate/McPkg.cs
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Domain/AggregateModels/InvitationAggregate/McPkg.cs:1]: Remove property SystemSubString. Not in use

File: src/Equinor.ProCoSys.IPO.Domain/AggregateModels/InvitationAggregate/McPkg.cs [hunk 14]
```diff
@@ -61,6 +69,7 @@ public McPkg(string plant, Project project, string commPkgNo, string mcPkgNo, st
         public ICollection<Certificate> CertificateScopes => _certificateScope;
         public DateTime CreatedAtUtc { get; private set; }
         public int CreatedById { get; private set; }
+        public Guid Guid { get; set; } = Guid.Empty;
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.Domain/AggregateModels/InvitationAggregate/McPkg.cs:72]: remove init Guid.Empty;
Add a Todo // make Guid private after FillGuids is completed

File: src/Equinor.ProCoSys.IPO.ForeignApi/MainApi/McPkg/IMcPkgApiService.cs [hunk 15]
```diff
@@ -7,6 +7,7 @@ public interface IMcPkgApiService
     {
         Task<IList<ProCoSysMcPkgOnCommPkg>> GetMcPkgsByCommPkgNoAndProjectNameAsync(string plant, string projectName, string commPkgNo);
         Task<IList<ProCoSysMcPkg>> GetMcPkgsByMcPkgNosAsync(string plant, string projectName, IList<string> mcPkgNos);
+        Task<ProCoSysMcPkg> GetMcPkgByIdAsync(string plant, long mcPkgNo);
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.ForeignApi/MainApi/McPkg/IMcPkgApiService.cs:10]: mcPkgId

File: src/Equinor.ProCoSys.IPO.ForeignApi/MainApi/McPkg/MainApiMcPkgService.cs [hunk 16]
```diff
@@ -42,6 +42,18 @@ public class MainApiMcPkgService : IMcPkgApiService
             return mcPkgs;
         }
 
+        public async Task<ProCoSysMcPkg> GetMcPkgByIdAsync(
+            string plant,
+            long mcPkgNo)
```
Comment (janid1967) [inline src/Equinor.ProCoSys.IPO.ForeignApi/MainApi/McPkg/MainApiMcPkgService.cs:47]: mcPkgId

File: src/tests/Equinor.ProCoSys.IPO.Command.Tests/InvitationCommands/AcceptPunchOut/AcceptPunchOutCommandHandlerTests.cs [hunk 17]
```diff
@@ -27,7 +27,8 @@ public class AcceptPunchOutCommandHandlerTests
         private AcceptPunchOutCommandHandler _dut;
         private const string _plant = "PCS$TEST_PLANT";
         private const string _projectName = "Project name";
-        private static readonly Project project = new(_plant, _projectName, $"Description of {_projectName} project");
+        private static readonly Guid _project1uid = new Guid("11111111-2222-2222-2222-333333333341");
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.Command.Tests/InvitationCommands/AcceptPunchOut/AcceptPunchOutCommandHandlerTests.cs:30]: _project1Guid

File: src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/InvitationAggregate/CommPkgTests.cs [hunk 18]
```diff
@@ -12,19 +12,21 @@ public class CommPkgTests
...
         private const string TestPlant = "PlantA";
         private const string ProjectName = "ProjectName";
+        private static readonly Guid ProjectGuid = new Guid("11111111-2222-2222-2222-333333333341");
         private const int ProjectId = 320;
         private const string CommPkgNo = "Comm1";
         private const string Description = "D1";
         private const string Status = "OK";
         private const string System = "1|2";
-        private readonly Project _project = new(TestPlant, ProjectName, $"Description of {ProjectName}");
+        private readonly Guid CommPkgGuid = Guid.Empty;
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/InvitationAggregate/CommPkgTests.cs:21]: make CommPkgGuid other than empty and Assert it is set in Constructor_ShouldSetProperties()

File: src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/InvitationAggregate/McPkgTests.cs [hunk 19]
```diff
@@ -12,19 +12,20 @@ public class McPkgTests
...
-        private readonly Project _project = new(TestPlant, ProjectName, $"Description of {ProjectName}");
+        private readonly Project _project = new(TestPlant, ProjectName, $"Description of {ProjectName}", ProjectGuid);
 
 
         [TestInitialize]
         public void Setup()
         {
             _project.SetProtectedIdForTesting(ProjectId);
-            _dut = new McPkg(TestPlant, _project, CommPkgNo, McPkgNo, Description, System);
+            _dut = new McPkg(TestPlant, _project, CommPkgNo, McPkgNo, Description, System, Guid.Empty);
```
Comment (janid1967) [inline src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/InvitationAggregate/McPkgTests.cs:28]: Add a private readonly McPkgGuid other than empty to set here, and Assert it is set in Constructor_ShouldSetProperties()

**Comment by janid1967** [comment]
Add a testclass for the Project domain class in src/tests/Equinor.ProCoSys.IPO.Domain.Tests/AggregateModels/InvitationAggregate/ Use McPkgTests.cs as template, just Setup and Constructor_ShouldSetProperties() needed

**Review by janid1967 — CHANGES_REQUESTED** [review]
See comments

**Review by janid1967 — APPROVED** [review]
Approved w minor improvements in CommPkgTests.cs and McPkgTests.cs

**Pull request merged**